mitodl/ocw-studio
# Option to delete existing pipelines in management commands

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #1391 

#### What's this PR do?
- Adds an optional parameter to all pipeline commands to delete existing pipelines first
- Modifies the concourse api, pipeline classes to support the above.

#### How should this be manually tested?
- Set up and run concourse locally as described in the README
- Run the following with and without the new `--delete` option, they all should run without errors.
`backpopulate_pipelines`
`upsert_site_removal_pipeline`
`upsert_theme_assets_pipeline`
`upsert_mass_build_pipeline`

It might be hard to notice if the pipelines are actually deleted before getting recreated, so if you want you can add a `return` line within the `if delete:` block of each command.


#### Any background context you want to provide?
There are lots of old orphaned site pipelines on RC.  One of them might be causing the draft S3 publish content to get deleted, not sure, but it would be good to remove these pipelines anyway.
